src/content/en/2021/resource-hints.md

```diff
@@ -1,20 +1,609 @@
+
+### The `media` attribute
+
+An oft-neglected feature available to `rel="preload"`, is the ability to specify media queries through the `media` attribute—with less than 4% of all preloads using this attribute. The `media` attribute accepts media queries allowing you to target the media type and specific browser features, such as viewport width. As an example, the `media` attribute would allow you to preload a low-resolution image on devices with a narrow viewport and a full-sized image on devices with a large viewport.
+
+In addition to the `media` attribute, the `<link>` element supports `imagesrcset` and `imagesizes` attributes which correspond to the `srcset` and `sizes` attributes on `<img>` elements. Using these attributes, you can use the same resource selection criteria that you would use on your image. Unfortunately, their adoption is very low (less than 1%); most likely owing to the lack of <a hreflang="en" href="https://caniuse.com/mdn-html_elements_link_imagesizes">support</a> on Safari.
+
+<p class="note">Note: The `media` attribute is not available on all `<link>` elements as the spec suggests, but it is only available on `rel="preload"`.</p>
+
+### Bad practices
+
+Owing to the versatility of `rel="preload"`, there isn't a clear set of rules dictating how to implement the preload hint, but we can learn a lot from our mistakes and understand how to avoid them.
+
+#### Unused preload
+
+We have already seen that there is a negative correlation between a website's performance and the number of preload hints. This relationship may be influenced by two factors:
+- Incorrect preloads
+- Unused preloads
+
+##### Incorrect preloads
+
+When you preload a resource which is not as important as the other resources which the browser would have otherwise prioritized. We are unable to measure the extent of incorrect preloads as you would need to A/B test the page with and without each hint.
+
+##### Unused preloads
+
+When you preload a resource which is not needed within the first few seconds of the page loading.
+
+{{ figure_markup(
+  caption='Percent of unused preload hints within the first 3 seconds.',
+  content="21.5%",
+  classes="big-number",
+  sheets_gid="2013605735",
+  sql_file="consoleLog_unused_preload.sql"
+)
+}}
+
+In such cases, the preload hint is regressing the website's performance, as you are instructing the browser to download and prioritize files or resources which are not needed immediately—or even not needed at all. This is one of the challenges when using resource hints, as they require regular maintenance and automating the process opens the door to allow such issues to creep in.
+
+{{ figure_markup(
```

Comment:
This figure doesn't seem to be discussed. Add a paragraph below with your takeaway.